rs/consensus/src/idkg.rs

```diff
@@ -488,111 +479,91 @@ impl IDkgPriorityFnArgs {
             active_transcripts.insert(transcript_ref.transcript_id);
         }
 
-        let (certified_height, requested_signatures) = state_reader
-            .get_certified_state_snapshot()
-            .map_or(Default::default(), |snapshot| {
-                let request_contexts = snapshot
-                    .get_state()
-                    .signature_request_contexts()
-                    .values()
-                    .flat_map(get_context_request_id)
-                    .collect::<BTreeSet<_>>();
+        let (certified_height, _) =
+            state_reader
+                .get_certified_state_snapshot()
+                .map_or(Default::default(), |snapshot| {
+                    let request_contexts = snapshot
+                        .get_state()
+                        .signature_request_contexts()
+                        .values()
+                        .flat_map(get_context_request_id)
+                        .collect::<BTreeSet<_>>();
 
-                (snapshot.get_height(), request_contexts)
-            });
+                    (snapshot.get_height(), request_contexts)
+                });
 
         Self {
             finalized_height: block_reader.tip_height(),
             certified_height,
-            requested_transcripts,
-            requested_signatures,
-            active_transcripts,
         }
     }
 }
 
 impl<Pool: IDkgPool> PriorityFnFactory<IDkgMessage, Pool> for IDkgGossipImpl {
-    fn get_priority_function(
-        &self,
-        _idkg_pool: &Pool,
-    ) -> PriorityFn<IDkgMessageId, IDkgMessageAttribute> {
+    fn get_priority_function(&self, _idkg_pool: &Pool) -> PriorityFn<IDkgMessageId> {
         let block_reader = IDkgBlockReaderImpl::new(self.consensus_block_cache.finalized_chain());
         let subnet_id = self.subnet_id;
         let args = IDkgPriorityFnArgs::new(&block_reader, self.state_reader.as_ref());
-        let metrics = self.metrics.clone();
-        Box::new(move |_, attr: &'_ IDkgMessageAttribute| {
-            compute_priority(attr, subnet_id, &args, &metrics)
-        })
+        Box::new(move |id| compute_priority(id, subnet_id, &args))
     }
 }
 
 fn compute_priority(
-    attr: &IDkgMessageAttribute,
+    id: &IDkgMessageId,
     subnet_id: SubnetId,
     args: &IDkgPriorityFnArgs,
-    metrics: &IDkgGossipMetrics,
 ) -> Priority {
-    match attr {
-        IDkgMessageAttribute::Dealing(transcript_id)
-        | IDkgMessageAttribute::DealingSupport(transcript_id) => {
+    match id {
+        IDkgMessageId::Dealing(_, data) => {
```

Comment:
Does keeping them together like before compile (same below)?
```suggestion
        IDkgMessageId::Dealing(_, data)
        | IDkgMessageId:: DealingSupport(_, data) => {
```